core/state/statedb.go

```diff
@@ -1450,9 +1532,17 @@ func (s *StateDB) Commit(failPostCommitFunc func(), postCommitFuncs ...func() er
 				}
 				// Only update if there's a state transition (skip empty Clique blocks)
 				if parent := s.snap.Root(); parent != s.expectedRoot {
-					if err := s.snaps.Update(s.expectedRoot, parent, s.snapDestructs, s.snapAccounts, s.snapStorage, verified); err != nil {
+					snapAccountLock.Lock()
```

Comment:
Lock `snapAccountLock` is not good idea, imaging `commmitTrie` get the lock first, then this routine have to wait a long time.